src/Controller/SpecialSponsorship/Edit.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\SpecialSponsorship;
+
+use App\Dto\SpecialSponsorshipRequest;
+use App\Form\SpecialSponsorshipType;
+use App\Repository\SpecialSponsorship\SpecialSponsorshipManagerInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\RouterInterface;
+use Twig_Environment;
+
+final class Edit
+{
+    private $renderer;
+    private $specialSponsorshipManager;
+    private $formFactory;
+    private $router;
+
+    public function __construct(
+        Twig_Environment $renderer,
+        SpecialSponsorshipManagerInterface $specialSponsorshipManager,
+        FormFactoryInterface $formFactory,
+        RouterInterface $router
+    )
+    {
+        $this->renderer = $renderer;
+        $this->specialSponsorshipManager = $specialSponsorshipManager;
+        $this->formFactory = $formFactory;
+        $this->router = $router;
+    }
+
+    /**
+     * @param Request $request
+     * @param string $id
+     * @return Response
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function handle(Request $request, string $id): Response
```

Comment:
I would prefer if you extracted the `$id` value from `$request->query->get('id');`, it would be nicer not to use the paramconverter here (too much magic, how do you go from a route to a param injected in your method?)
Also the paramconverter is part of `sensio/framework-extra-bundle`, which should ultimately be removed as it comes with way too many features at once (annotation, param converter, event listeners and whatnot).